lib/observer.js

```diff
@@ -231,7 +231,15 @@ ObserverMixin.notifyObserversAround = function(operation, context, fn, callback)
 
       function cbForWork(err) {
         const args = [].slice.call(arguments, 0);
-        if (err) return callback.apply(null, args);
+        if (err) {
+          // call observer in case of error to hook response
+          context.error = err;
+          self.notifyObserversOf('after ' + operation + ' error', context,
+            function(_err, context) {
+              callback.call(null, _err || err, context);
```

Comment:
I am not sure if it's a good idea to give priority to the error reported by the hook over the actual error in the database operation. @raymondfeng @strongloop/loopback-maintainers thoughts?

Either way, we can report only one error and the other error will be discarded. To make troubleshooting easier, it's important to print the discarded error to the debug log.

E.g.

```js
if (_err && err) {
  debug(
    'Operation %j failed and "after %s error" hook returned an error too. ' +
      'Calling back with the hook error only.' +
      '\nOriginal error: %s\nHook error: %s\n',
    err.stack || err,
    _err.stack || _err);
}
```